ghz_fpga_server.py

```diff
@@ -835,61 +835,44 @@ def extractTiming(self, packets):
     def recoverFromTimeout(self, runners, results):
         """Recover from a timeout error so that pipelining can proceed.
 
-        We clear the packet buffer for all boards, whether or not they
-        succeeded. We then check how many time each board executed its SRAM
-        sequence (or demod sequence for ADC boards), and store this value
-        in the respective runner object. Finally, for failed boards we
-        send a trigger to the master context.
+        The recovery proceeds as follows: first, we clear the packet buffer for
+        each board and ping it to see how many times it executed its SRAM
+        sequence (or demod sequence for ADC boards). This count is stored in
+        the runner object for the board for later reporting to the user.
+
+        After all boards have been pinged, we again clear the packet buffers
+        for all boards and then send a trigger to the board group's run context
+        from each failed board (since the trigger would not have been sent yet
+        if the packet collection failed).
         """
         print 'RECOVERING FROM TIMEOUT'
-        # TODO: add a timeout to the read. Possibly use _sendRegister from
-        #       dac.py
         for runner, (success, result) in zip(runners, results):
-            ctx = None if success else self.ctx
-            # 1. Clear packet buffer for this board
             yield runner.dev.clear().send()
-            # 2. Ping the board to read its execution counter
             try:
+                # NOTE: in the current implementation of regPing for DAC boards
+                # (build 15) the start field is set to master, which means when
+                # we ping these boards they will emit daisy chain signals.
                 p = runner.dev.regPingPacket()
-                p.timeout(U.Value(1.0, 's')).collect(1).read(1)
+                p.timeout(U.Value(1.0, 's')).read(1)
```

Comment:
I'd chain this to the previous line, or write three separate lines, i.e. keep it consistent.